src/funnels/Depositor.sol

```diff
@@ -0,0 +1,275 @@
+// SPDX-FileCopyrightText: © 2023 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.16;
+
+import {LiquidityAmounts} from "src/funnels/uniV3/LiquidityAmounts.sol";
+import {TickMath}         from "src/funnels/uniV3/TickMath.sol";
+
+interface RolesLike {
+    function canCall(bytes32, address, address, bytes4) external view returns (bool);
+}
+
+interface GemLike {
+    function transferFrom(address, address, uint256) external;
+}
+
+// https://github.com/Uniswap/v3-core/blob/d8b1c635c275d2a9450bd6a78f3fa2484fef73eb/contracts/UniswapV3Pool.sol
+interface UniV3PoolLike {
```

Comment:
In theory we should remove all the variable names from here but makes me wonder if we shouldn't keep them.

Reply:
We need them for the named params.

Maybe that should be the rule of thumb - that we keep the param names only when the interface has at least one function that we use named params when calling.